kubernetes/resource_kubernetes_service_account.go

```diff
@@ -0,0 +1,220 @@
+
+	namespace, name := idParts(d.Id())
+	log.Printf("[INFO] Reading service account %s", name)
+	svcAcc, err := conn.CoreV1().ServiceAccounts(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		log.Printf("[DEBUG] Received error: %#v", err)
+		return err
+	}
+	log.Printf("[INFO] Received service account: %#v", svcAcc)
+	err = d.Set("metadata", flattenMetadata(svcAcc.ObjectMeta))
+	if err != nil {
+		return err
+	}
+	d.Set("image_pull_secret", flattenLocalObjectReferenceArray(svcAcc.ImagePullSecrets))
+
+	defaultSecretName := d.Get("default_secret_name").(string)
+	log.Printf("[DEBUG] Default secret name is %q", defaultSecretName)
+	secrets := flattenServiceAccountSecrets(svcAcc.Secrets, defaultSecretName)
+	log.Printf("[DEBUG] Flattened secrets: %#v", secrets)
+	d.Set("secret", secrets)
+
+	return nil
+}
+
+func resourceKubernetesServiceAccountUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	namespace, name := idParts(d.Id())
+
+	ops := patchMetadata("metadata.0.", "/metadata/", d)
+	if d.HasChange("image_pull_secret") {
+		v := d.Get("image_pull_secret").(*schema.Set).List()
+		ops = append(ops, &ReplaceOperation{
+			Path:  "/imagePullSecrets",
+			Value: expandLocalObjectReferenceArray(v),
+		})
+	}
+	if d.HasChange("secret") {
+		v := d.Get("secret").(*schema.Set).List()
+		defaultSecretName := d.Get("default_secret_name").(string)
+
+		ops = append(ops, &ReplaceOperation{
+			Path:  "/secrets",
+			Value: expandServiceAccountSecrets(v, defaultSecretName),
+		})
+	}
+	data, err := ops.MarshalJSON()
+	if err != nil {
+		return fmt.Errorf("Failed to marshal update operations: %s", err)
+	}
+	log.Printf("[INFO] Updating service account %q: %v", name, string(data))
+	out, err := conn.CoreV1().ServiceAccounts(namespace).Patch(name, pkgApi.JSONPatchType, data)
+	if err != nil {
+		return fmt.Errorf("Failed to update service account: %s", err)
+	}
+	log.Printf("[INFO] Submitted updated service account: %#v", out)
+	d.SetId(buildId(out.ObjectMeta))
+
+	return resourceKubernetesServiceAccountRead(d, meta)
+}
+
+func resourceKubernetesServiceAccountDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	namespace, name := idParts(d.Id())
+	log.Printf("[INFO] Deleting service account: %#v", name)
+	err := conn.CoreV1().ServiceAccounts(namespace).Delete(name, &metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Service account %s deleted", name)
+
+	d.SetId("")
+	return nil
+}
+
+func resourceKubernetesServiceAccountExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	conn := meta.(*kubernetes.Clientset)
+
+	namespace, name := idParts(d.Id())
+	log.Printf("[INFO] Checking service account %s", name)
+	_, err := conn.CoreV1().ServiceAccounts(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		if statusErr, ok := err.(*errors.StatusError); ok && statusErr.ErrStatus.Code == 404 {
```

Comment:
Aye, we could do and under many circumstances I would, I just feel that for common codes like 200 and 404 it's just not worth it as three digits are shorter and nearly everyone knows what these mean.